.github/workflows/gh-pages.yml

```diff
@@ -1,53 +1,57 @@
 name: gh-pages
 
 on:
+  pull_request:
+    branches:
+      - master
   push:
     branches:
       - master
   schedule:
     - cron: '30 1 * * *'
   workflow_dispatch:
 
-permissions:
-  contents: write
-
 jobs:
   translations:
     runs-on: ubuntu-latest
     if: (github.event_name == 'schedule' && github.repository == 'opengisch/QField-docs') || (github.event_name != 'schedule')
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+        uses: docker/setup-buildx-action@v3
 
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: '3.10'
           cache: 'pip'
 
       - name: Install Python requirements
-        run: pip install -r requirements.txt
+        run: |
+          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
 
-      # - name: Install Python requirements insiders
-      #   run: pip install -r requirements-insiders.txt
-      #   env:
-      #     MKDOCS_INSIDERS_TOKEN: ${{ secrets.MKDOCS_INSIDERS_TOKEN }}
+      - name: Install Python requirements insiders
+        run: pip install -r requirements-insiders.txt
+        env:
+          MKDOCS_INSIDERS_TOKEN: ${{ secrets.MKDOCS_INSIDERS_TOKEN }}
 
       - name: Install Transifex client
         run: |
-          curl -OL https://github.com/transifex/cli/releases/download/v1.3.1/tx-linux-amd64.tar.gz
+          curl -OL https://github.com/transifex/cli/releases/download/v1.6.10/tx-linux-amd64.tar.gz
           tar -xvzf tx-linux-amd64.tar.gz
 
+      - name: Extract translatable content from mkdocs.yml config
+        run: ./utils/mkdocs_tx.py create_source
+
       - name: Configure Transifex
-        run: |
-          python ./utils/transifex_utils.py
+        run: ./utils/transifex_utils.py
         env:
           TX_TOKEN: ${{ secrets.TX_TOKEN }}
 
       - name: Push source files to Transifex
-        run: |
-          ./tx push
+        if: ${{ github.event_name != 'pull_request' }}
```

Comment:
I think it is safer if we just list the `event_name`s that we want to run step on. Otherwise if we change the GH Actions `on` section, it might run something we don't want.